tests/trainer/test_sharded_checkpoint.py

```diff
@@ -293,50 +333,58 @@ def test_fsdp_full_state_dict_load_with_ema(world_size, tmp_path: pathlib.Path,
 
 @pytest.mark.gpu
 @world_size(2)
+@pytest.mark.parametrize('weights_only', [True, False])
 @pytest.mark.parametrize('optimizer', ['adam', 'adamw'])
 @pytest.mark.parametrize('state_dict_type', ['local', 'sharded'])
 @pytest.mark.parametrize('precision', ['amp_bf16', 'amp_fp16'])
-@pytest.mark.parametrize('autoresume', [True, False])
+@pytest.mark.parametrize('autoresume', [False])  # True commented out for now
 @pytest.mark.skipif(version.parse(torch.__version__) < version.parse('1.13.0'),
                     reason='requires PyTorch 1.13 or higher')
+@pytest.mark.filterwarnings(r'ignore:TypedStorage is deprecated.:UserWarning')
 def test_fsdp_partitioned_state_dict_load(world_size, tmp_path: pathlib.Path, state_dict_type: str, autoresume: bool,
-                                          precision: str, optimizer: str):
+                                          precision: str, optimizer: str, weights_only: bool):
+    if state_dict_type == 'local':
+        pytest.xfail(
+            'Loading a state_dict_type="local" checkpoint with strict=True errors out. See https://github.com/pytorch/pytorch/issues/102667 for more info'
+        )
     if autoresume:
         run_name = 'my-autoresume-run'
     else:
         run_name = None
-    save_folder = tmp_path
+    save_folder = '/tmp/test_checkpoints'
```

Comment:
how come this change to hardcode tmp path?

Reply:
all ranks have to share the same folder because they all need the .metadata file. If you use tmp_path then each rank gets a different path

Reply:
yeah, i think it is better to sync the tmp path from rank 0, so that the dir is auto cleaned up